setup.cfg

```diff
@@ -4,3 +4,9 @@ universal=1
 [metadata]
 project_urls =
     Changelog = https://github.com/AzureAD/microsoft-authentication-library-for-python/releases
+
+[options.extras_require]
+# The broker is defined as optional dependency, so that downstream apps can opt in.
+# The opt-in is needed because historically most MSAL Python apps do not have the redirect_uri needed by broker
+#broker =
+#  pymsalruntime>=0.3,<0.6;python_version>='3' and platform_system=='Windows'
```

Comment:
>>=0.3,<0.6 [](http://example.com/codeflow?start=16&length=10)

Currently we are at 0.5 and we will rev the version once the removing select_account work is done. So does it make sense to set it <= 0.6?
And does 0.6 includes 0.6.X ?

Reply:
> Currently we are at 0.5 and we will rev the version once the removing select_account work is done. So does it make sense to set it <= 0.6?

No, that would defeat the purpose of managing dependencies by their versioning. The current approach is correct: MSAL Python is written on top of the `pymsalruntime >=0.3,<=0.5.x`'s behavior and declare that we will NOT pull in pymsalruntime 0.6+. When and only when pymsalruntime 0.6 becomes available, MSAL Python will make relevant changes in the business logic AND bump the version requirement.

> And does 0.6 includes all versions within 0.6.X ?
It is declared as `<0.6`, so it means all versions with 0.5.X.

P.S.: [Semantic Versioning](https://semver.org/) is the versioning scheme all MSALs use. We strongly recommend PyMsalRuntime would also use this scheme.